### ISSUE TITLE
returning address of local temporary object 

### ISSUE THREAD

**Issue by Areustle** [fermi-lat/tip#6]
Uh oh, we're undefining all the behaviors!!!

```

In file included from /Users/areustle/nasa/fermipy3/include/tip/IFileSvc.h:14:
/Users/areustle/nasa/fermipy3/include/tip/Header.h:146:65: warning: returning address of local temporary object [-Wreturn-stack-address]
      virtual Iterator begin() { unsupported("begin()"); return KeySeq_t().begin(); }
                                                                ^~~~~~~~~~
/Users/areustle/nasa/fermipy3/include/tip/Header.h:148:61: warning: returning address of local temporary object [-Wreturn-stack-address]
      virtual Iterator end() { unsupported("end()"); return KeySeq_t().end(); }
                                                            ^~~~~~~~~~
/Users/areustle/nasa/fermipy3/include/tip/Header.h:150:82: warning: returning address of local temporary object [-Wreturn-stack-address]
      virtual ConstIterator begin() const { unsupported("begin() const"); return KeySeq_t().begin(); }
                                                                                 ^~~~~~~~~~
/Users/areustle/nasa/fermipy3/include/tip/Header.h:152:78: warning: returning address of local temporary object [-Wreturn-stack-address]
      virtual ConstIterator end() const { unsupported("end() const"); return KeySeq_t().end(); }
                                                                             ^~~~~~~~~~
/Users/areustle/nasa/fermipy3/include/tip/Header.h:158:101: warning: returning address of local temporary object [-Wreturn-stack-address]
      virtual Iterator find(const std::string &) { unsupported("find(const std::string &)"); return KeySeq_t().end(); }
                                                                                                    ^~~~~~~~~~
/Users/areustle/nasa/fermipy3/include/tip/Header.h:165:66: warning: returning address of local temporary object [-Wreturn-stack-address]
        { unsupported("find(const std::string &) const"); return KeySeq_t().end(); }
                                                                 ^~~~~~~~~~
/Users/areustle/nasa/fermipy3/include/tip/Header.h:178:101: warning: returning address of local temporary object [-Wreturn-stack-address]
      virtual Iterator append(const KeyRecord &) { unsupported("append(const KeyRecord &)"); return KeySeq_t().end(); }
                                                                                                    ^~~~~~~~~~
/Users/areustle/nasa/fermipy3/include/tip/Header.h:184:81: warning: returning address of local temporary object [-Wreturn-stack-address]
      virtual Iterator erase(Iterator) { unsupported("erase(Iterator)"); return KeySeq_t().end(); }
                                                                                ^~~~~~~~~~
8 warnings generated.
```

**Comment by Areustle** [comment]
The issue occurs only on clang in tip::Header class Header. This is an impure virtual container class which  holds a map of keywords. The class also exposes an iterator over a "Key Sequence" type which is typedef-ed as a vector of KeyRecords (not keywords) but has no data member. 

In the immediate term I can "fix" this by replacing these local temporaries with constructors of empty (invalid) iterators. the existing iterators are already de-facto invalid since they point to local temporaries. 

A longer term fix would be to rewrite the iterator style interface of the Header container to expose the std::map::iterator types over the keyword map, rather than devoting other memory in the dependent classes to a sequence vector with redundant data.

**Comment by Areustle** [comment]
Fixed by PR #7
